File: AdminBackend/AdminBackend/src/AdminBackend.Application/Services/DataManager/JobRequest.cs
using System.Text.Json.Serialization;

namespace AdminBackend.Application.Services.DataManager;

public record JobRequest
{
    [JsonPropertyName("application_id")]
    public required long ApplicationId { get; init; }
    [JsonPropertyName("file_id")]
    public required long FileId { get; init; }
    [JsonPropertyName("file_store_id")]
    public required long FileStoreId { get; init; }
    [JsonPropertyName("document_id")]
    public required string DocumentId { get; init; }
    [JsonPropertyName("file_extension")]
    public required string FileExtension { get; init; }
    [JsonPropertyName("bucket_name")]
    public string? BucketName { get; init; }
    [JsonPropertyName("storage_type")]
    public required string StorageType { get; init; }  // s3
    [JsonPropertyName("extractor_type")]
    public required string ExtractorType { get; init; } // markitdown
    [JsonPropertyName("chunk_type")]
    public required string ChunkType { get; init; } // markdown, fixed-size, html
    [JsonPropertyName("chunk_size")]
    public required int ChunkSize { get; init; }
    [JsonPropertyName("chunk_overlap")]
    public required int ChunkOverlap { get; init; }
}

File: DataManager/src/models/job_request.py
from pydantic import BaseModel

from src.chunking.splitter_types import SplitterTypes
from src.extraction.extractor_types import ExtractorTypes
from src.storage.storage_types import StorageTypes


class JobRequest(BaseModel):
    application_id: int
    file_id: int
    file_store_id: int
    document_id: str
    file_extension: str
    bucket_name: str | None
    storage_type: StorageTypes
    extractor_type: ExtractorTypes
    chunk_type: SplitterTypes
    chunk_size: int
    chunk_overlap: int
